pkg/plugin/aggregation/run.go

```diff
@@ -190,9 +192,14 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 		return errors.Wrapf(err, "couldn't get aggregator pod")
 	}
 
-	for _, p := range plugins {
-		logrus.WithField("plugin", p.GetName()).Info("Running plugin")
-		go aggr.RunAndMonitorPlugin(context.Background(), time.Duration(cfg.TimeoutSeconds)*time.Second, p, client, nodes.Items, cfg.AdvertiseAddress, certs[p.GetName()], aggregatorPod, progressPort, pluginResultsDir)
+	for _, pluginGroup := range getOrderedPlugins(plugins) {
+		var waitGroup sync.WaitGroup
```

Comment:
👍 to using a WaitGroup. As someone new to Go I'm really happy to see you explored the options and found this. It really simplifies the logic.